atlasdb-workload-server/src/test/java/com/palantir/atlasdb/workload/workflow/SingleBusyCellWorkflowsTest.java

```diff
@@ -0,0 +1,101 @@
+            ImmutableSingleBusyCellWorkflowConfiguration.builder()
+                    .tableConfiguration(ImmutableTableConfiguration.builder()
+                            .tableName(TABLE_NAME)
+                            .isolationLevel(IsolationLevel.SERIALIZABLE)
+                            .build())
+                    .iterationCount(ITERATION_COUNT)
+                    .build();
+
+    private final InteractiveTransactionStore transactionStore = AtlasDbTransactionStore.create(
+            TransactionManagers.createInMemory(ImmutableSet.of()),
+            ImmutableMap.of(
+                    TableReference.createWithEmptyNamespace(TABLE_NAME),
+                    AtlasDbUtils.tableMetadata(IsolationLevel.SERIALIZABLE)));
+    private final Workflow workflow = SingleBusyCellWorkflows.create(
+            transactionStore,
+            CONFIGURATION,
+            MoreExecutors.listeningDecorator(PTExecutors.newSingleThreadExecutor()),
+            MoreExecutors.listeningDecorator(PTExecutors.newSingleThreadExecutor()));
+
+    @Test
+    public void workflowHistoryTransactionStoreShouldBeReadOnly() {
+        WorkflowHistory history = workflow.run();
+        assertThat(history.transactionStore()).isInstanceOf(ReadOnlyTransactionStore.class);
+    }
+
+    @Test
+    public void transactionsDoNotConflictExcessively() {
+        WorkflowHistory workflowHistory = workflow.run();
+        assertThat(workflowHistory.history()).hasSizeBetween(ITERATION_COUNT / 2, ITERATION_COUNT);
+    }
+
+    @Test
+    public void successfulTransactionsOnlyReadOrModifyTheTargetCell() {
+        WorkflowHistory workflowHistory = workflow.run();
+        List<WitnessedTransactionAction> witnessedTransactionActions = workflowHistory.history().stream()
+                .map(WitnessedTransaction::actions)
+                .flatMap(Collection::stream)
+                .collect(Collectors.toList());
+        assertThat(witnessedTransactionActions)
+                .allMatch(action -> action.table().equals(TABLE_NAME));
+        assertThat(witnessedTransactionActions)
+                .allMatch(action -> action.cell().equals(SingleBusyCellWorkflows.BUSY_CELL));
+    }
+
+    @Test
+    public void targetCellIsReadModifiedAndDeleted() {
+        WorkflowHistory workflowHistory = workflow.run();
+        List<WitnessedTransactionAction> witnessedTransactionActions = workflowHistory.history().stream()
+                .map(WitnessedTransaction::actions)
+                .flatMap(Collection::stream)
+                .collect(Collectors.toList());
+        assertThat(witnessedTransactionActions).anyMatch(action -> action instanceof WitnessedReadTransactionAction);
+        assertThat(witnessedTransactionActions).anyMatch(action -> action instanceof WitnessedWriteTransactionAction);
+        assertThat(witnessedTransactionActions).anyMatch(action -> action instanceof WitnessedDeleteTransactionAction);
```

Comment:
nit: `WitnessedDeleteTransactionAction.class::instanceOf` ?

Reply:
I think you mean `isInstance`, but yeah well spotted 👍